src/dxorshift/wrapper.d

```diff
@@ -0,0 +1,401 @@
+ *
+ * Example:
+ * --------
+ * import std.random : Random;
+ * import std.range : take;
+ *
+ * import dxorshift.xoroshiro128plus : Xoroshiro128plus;
+ * import dxorshift.wrapper : uniformRNG;
+ *
+ * auto gen = Xoroshiro128plus(123456);
+ *
+ * // RNGs defined in the dxorshift package
+ * // cannot be copied by value.  Passing
+ * // them via pointer like this is not
+ * // permitted in `@safe` code:
+ * auto unsafeTake = (&gen).take(10);
+ *
+ * // however, the `uniformRNG` wrapper can
+ * // be used in `@safe` code:
+ * auto safeTake = gen.uniformRNG.take(10);
+ *
+ * // it can also be used to wrap generators
+ * // from phobos' `std.random` to guarantee
+ * // they will have reference-type, input
+ * // range semantics
+ * auto gen2 = Random(123456);
+ * auto phobosTake = gen2.uniformRNG.take(10);
+ * --------
+ *
+ * Authors:
+ *     $(LINK2 http://braingam.es/, Joseph Rushton Wakeling)
+ *
+ * Copyright:
+ *     Written in 2016 by Joseph Rushton Wakeling.
+ *
+ * License:
+ *     $(LINK2 https://creativecommons.org/publicdomain/zero/1.0/legalcode, Creative Commons CC0)
+ *     (public domain)
+ */
+module dxorshift.wrapper;
+
+import std.random : isUniformRNG;
+
+/**
+ * Safety wrapper that ensures reference-type, input-range
+ * access to the underlying uniform random number generator.
+ *
+ * This can be used for two distinct purposes.  First, it
+ * can be used to wrap RNGs whose postblit has been disabled
+ * (such as those in dxorshift), allowing them to be used in
+ * `@safe` code; second, it can be used to wrap random number
+ * generators from Phobos' `std.random` in a way that should
+ * prevent the main sources of unintended copying of the RNG
+ * (both direct copy-by-value, and use of the `save` property).
+ *
+ * Template_Params:
+ *     BaseRNG = underlying type of uniform random number
+ *               generator
+ */
+public struct SafeUniformRNG (BaseRNG)
+    if (isUniformRNG!BaseRNG)
+{
+  public:
+    /**
+     * Constructor
+     *
+     * Params:
+     *     gen = uniform random number generator instance
+     *           to wrap; its lifetime must be guaranteed
+     *           throughout the corresponding lifetime of
+     *           the `SafeUniformRNG` instance
+     */
+    this (return ref BaseRNG gen) @trusted
+    {
+        this.generator = &gen;
+    }
+
+    /// Marks this range as a uniform random number generator
+    enum isUniformRandom = BaseRNG.isUniformRandom;
+
+    /// Smallest generated value
+    enum min = BaseRNG.min;
+
+    /// Largest generated value
+    enum max = BaseRNG.max;
+
+    static if (isInfinite!BaseRNG)
+    {
+        /// Range primitives
+        enum bool empty = false;
+    }
+    else
+    {
+        // unlikely that any RNG type will not
+        // be an infinite range, but let's at
+        // least take care of the possibility
+
+        /// Range primitives
+        bool empty() @property
+        {
+            return this.generator.empty;
+        }
+    }
```

Comment:
Genuinely not sure if the above is a generalization too far; TBH, as with `min` and `max`, I think it may be best to assume `BaseRNG.empty` is also an enum, and just use that.

The moment someone comes along with a D RNG implementation where `empty` is not an `enum` equal to `false`, I might reconsider ... ;-)